modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -115,89 +364,865 @@ public void testGetSuccess(String id, String attributes,
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+        User userOld = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING, schema, new User());
+        String id = userOld.getId();
+        return new Object[][]{
+                {id, NEW_USER_SCIM_OBJECT_STRING_UPDATE, "userName", null, ResponseCodeConstants.CODE_BAD_REQUEST}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestUpdateWithPUTBadRequestException")
+    public void testUpdateWithPUTBadRequestException(String id, String scimObjectString, String
+            attributes, String excludeAttributes, int expectedScimResponseStatus) {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT);
+        abstractResourceManager.when(()
+                -> AbstractResourceManager.encodeSCIMException(any(BadRequestException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new BadRequestException()));
+        abstractResourceManager.when(() -> userManager.getUser(id,
+                ResourceManagerUtil.getAllAttributeURIs(schema))).thenThrow(BadRequestException.class);
+        SCIMResponse scimResponse = userResourceManager.updateWithPUT(id, scimObjectString, userManager,
+                attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForListWithPOST")
+    public Object[][] dataToListWithPOST() throws BadRequestException, CharonException, InternalErrorException {
+
+        List<Object> tempList = new ArrayList<>();
+        tempList.add(1);
+        tempList.add(getNewUser());
+        return new Object[][]{
+                {RESOURCE_STRING, tempList}
+        };
+    }
+
+    @Test(dataProvider = "dataForListWithPOST")
+    public void testListWithPOST(String resourceString, List<Object> tempList)
+            throws NotImplementedException, BadRequestException, CharonException {
+
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT + "/.search");
+        Mockito.when(userManager.listUsersWithPost(any(SearchRequest.class), anyMap())).thenReturn(tempList);
+        SCIMResponse scimResponse = userResourceManager.listWithPOST(resourceString, userManager);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_OK);
+    }
+
+    @Test
+    public void testListWithPOSTCharonException()
+            throws NotImplementedException, BadRequestException, CharonException {
```

Comment:
merge this into 1 line

Reply:
Fixed in commit 504369f8bdb4f024c63ae5d78f16b04decbedc11